### PR TITLE
Bump MLIR version

### PR DESCRIPTION
**Pull request by nmostafa**
* Fixes in ngraph-opt
* Fix JIT invocation ABI to use another level of indirection for args.

### CODE REVIEW

File: src/contrib/mlir/compiler/compiler.cpp [hunk 1]
```diff
@@ -729,11 +736,10 @@ void MLIRCompiler::bindArguments(std::vector<void*>& externalTensors)
     m_externalTensors = &externalTensors;
 
     // Create list with a type-erased double pointer for each invocation arguments.
-    // We currently use 'allocateMemrefArgs', which creates a
-    // SmallVector<StaticFloatMemref*>. StaticFloatMemref is just a struct with the
-    // actual pointer to the data.
+    // We currently use 'allocateMemrefArgs', which creates the arguments list per call ABI (see
+    // comment below).
+    // StaticFloatMemref is just a struct with the actual pointer to the data.
```
Comment (dcaballe) [inline src/contrib/mlir/compiler/compiler.cpp]: Not sure I can understand the `StaticFloatMemref` comment now. Could we elaborate this a bit more, add more info about the ABI here or rephrase this two comments in some way?

File: src/contrib/mlir/compiler/compiler.cpp [hunk 2]
```diff
@@ -744,7 +750,8 @@ void MLIRCompiler::bindArguments(std::vector<void*>& externalTensors)
     // Assign external tensor pointers to invocation arguments.
     for (size_t i = 0, numArgs = m_invokeArgs.size(); i < numArgs; ++i)
     {
-        ((mlir::StaticFloatMemRef*)m_invokeArgs[i])->data = (float*)(*m_externalTensors)[i];
+        auto* memRefArg = *((mlir::StaticFloatMemRef**)m_invokeArgs[i]);
+        memRefArg->data = (float*)(*m_externalTensors)[i];
```
Comment (dcaballe) [inline src/contrib/mlir/compiler/compiler.cpp]: Use C++ castings instead?

File: test/backend/dot.in.cpp [hunk 3]
```diff
@@ -254,7 +254,7 @@ NGRAPH_TEST(${BACKEND_NAME}, dot_0_0)
     EXPECT_TRUE(test::all_close_f((vector<float>{0}), read_vector<float>(result)));
 }
 
-NGRAPH_TEST(${BACKEND_NAME}, dot_matrix_2x0_0x2)
+NGRAPH_TEST(${BACKEND_NAME}, MLIR_DISABLE_TEST(dot_matrix_2x0_0x2))
```
Comment (dcaballe) [inline test/backend/dot.in.cpp]: Please, add this test to NGCPU-461

File: src/contrib/mlir/compiler/compiler.cpp (hunk 1 above)
Comment (nmostafa) [reply src/contrib/mlir/compiler/compiler.cpp]: 790 has an example of the expected args layout in memory.

**Comment by nmostafa** [comment]
CI MLIR tests pass with [MLIR] tag in title, but because MLIR repo commit msg below containing `error` keyword, CI thinks it is failing. 
`HEAD is now at 1f7893e0 Fix Windows linkage error`

Removing [MLIR] tag to be able to merge. 
@aslepko for reference.

**Pull request merged**